### ISSUE TITLE
How does TaskQueue/TaskWorklet tie into WASM threads?

### ISSUE THREAD

**Issue by pshihn** [developit/task-worklet#9]
With threads in WASM coming up, there may be some merit to tying them with Task Queues

**Comment by developit** [comment]
I think right now there isn't really an interaction between the two. The polyfill would allow WASM as it is implemented using Workers for the threadpool, however the proposal (publishing soon) and our experimental implementation in Chromium use a proper Worklet backing in the pool and thus don't allow for WASM usage.

**Comment by chicoxyzzy** [comment]
It's possible to use WebAssembly in AudioWorklet in stable Chrome so it's confusing that it's not the case for TaskWorklet

**Comment by chicoxyzzy** [comment]
Basically it should be possible to load js module via `Worklet.addModule` and load+compile+instantiate wasm in this module

**Comment by chicoxyzzy** [comment]
Also just found [this reference](https://webassembly.github.io/spec/js-api/#dictdef-webassemblyinstantiatedsource) that `WebAssembly` namespace should be exposed to window, worker and worklet

**Comment by developit** [comment]
@chicoxyzzy you're right, I'm not sure why I implied WASM couldn't be used - there's no reason it can't here.

```js
const wasm = WebAssembly.instantiateStreaming(fetch('worklet.wasm'));

class WasmDemoProcessor {
  async process(data) {
    const { instance } = await wasm;
    return instance.exports.doThing(data);
  }
}

registerTask('wasm-demo', WasmDemoProcessor);
```
